addons/resource/models/resource.py

```diff
@@ -536,193 +478,175 @@ def _schedule_days(self, cr, uid, id, days, day_date=None, compute_leaves=False,
+        :deprecated: Odoo saas-3. Use get_working_hours instead."""
+        return self._interval_hours_get(dt_from, dt_to, resource_id=resource)
 
-        :deprecated: OpenERP saas-3. Use get_working_hours instead."""
-        return self._interval_hours_get(cr, uid, id, dt_from, dt_to, resource_id=resource)
-
-    def _interval_hours_get(self, cr, uid, id, dt_from, dt_to, resource_id=False, timezone_from_uid=None, exclude_leaves=True, context=None):
+    @api.model
+    def _interval_hours_get(self, dt_from, dt_to, resource_id=False, timezone_from_uid=None, exclude_leaves=True):
         """ Computes working hours between two dates, taking always same hour/minuts.
-
-        :deprecated: OpenERP saas-3. Use get_working_hours instead. Note: since saas-3,
+        :deprecated: Odoo saas-3. Use get_working_hours instead. Note: since saas-3,
         now resets hour/minuts. Now counts leave hours instead of all-day leaves."""
-        return self.get_working_hours(
-            cr, uid, id, dt_from, dt_to,
-            compute_leaves=(not exclude_leaves), resource_id=resource_id,
-            default_interval=(8, 16), context=context)
+        return self.get_working_hours(dt_from, dt_to,
+                                      compute_leaves=(not exclude_leaves), resource_id=resource_id,
+                                      default_interval=(8, 16))
 
 
-class resource_calendar_attendance(osv.osv):
+class ResourceCalendarAttendance(models.Model):
     _name = "resource.calendar.attendance"
     _description = "Work Detail"
-
-    _columns = {
-        'name' : fields.char("Name", required=True),
-        'dayofweek': fields.selection([('0','Monday'),('1','Tuesday'),('2','Wednesday'),('3','Thursday'),('4','Friday'),('5','Saturday'),('6','Sunday')], 'Day of Week', required=True, select=True),
-        'date_from' : fields.date('Starting Date'),
-        'date_to': fields.date('End Date'),
-        'hour_from' : fields.float('Work from', required=True, help="Start and End time of working.", select=True),
-        'hour_to' : fields.float("Work to", required=True),
-        'calendar_id' : fields.many2one("resource.calendar", "Resource's Calendar", required=True),
-    }
-
     _order = 'dayofweek, hour_from'
 
-    _defaults = {
-        'dayofweek' : '0'
-    }
+    name = fields.Char(required=True)
+    dayofweek = fields.Selection([('0', 'Monday'), ('1', 'Tuesday'), ('2', 'Wednesday'), ('3', 'Thursday'), ('4', 'Friday'), ('5', 'Saturday'), ('6', 'Sunday')], string='Day of Week', default='0', required=True, index=True)
+    date_from = fields.Date(string='Starting Date')
+    date_to = fields.Date(string='End Date')
+    hour_from = fields.Float(
+        string='Work from', required=True, help="Start and End time of working.", index=True)
+    hour_to = fields.Float(string='Work to', required=True)
+    calendar_id = fields.Many2one(
+        "resource.calendar", string="Resource's Calendar", required=True)
+
 
 def hours_time_string(hours):
```

Comment:
What is use of this method ? And **if the method is really necessary**, place it after imports as this is a global method.